timeseries/src/autogluon/timeseries/utils/features.py

```diff
@@ -284,3 +300,53 @@ def _check_required_columns_are_present(
             raise ValueError(
                 f"{len(missing_columns)} columns are missing from {data_frame_name}: {reprlib.repr(missing_columns.to_list())}"
             )
+
+
+class AbstractFeatureImportanceTransform:
+    """Abstract class for transforms that replace a given feature with dummy or shuffled values,
+    for use in feature importance operations.
+    """
+
+    def __init__(
+        self,
+        covariate_metadata: CovariateMetadata,
+        **kwargs,
+    ):
+        self.covariate_metadata: CovariateMetadata = covariate_metadata
+
+    def _apply_transform(self, data: TimeSeriesDataFrame, feature_name: str) -> TimeSeriesDataFrame:
+        raise NotImplementedError
+
+    def transform(self, data: TimeSeriesDataFrame, feature_name: str, **kwargs) -> TimeSeriesDataFrame:
+        if feature_name not in self.covariate_metadata.all_features:
+            raise ValueError(f"Target feature {feature_name} not found in covariate metadata")
+
+        # feature transform works on a shallow copy of the main time series data frame
+        # but a deep copy of the static features.
+        data = data.copy(deep=False)
+
+        return self._apply_transform(data, feature_name, **kwargs)
+
+
+class PermutationFeatureImportanceTransform(AbstractFeatureImportanceTransform):
+    """Naively shuffles a given feature."""
+
+    def _apply_transform(self, data: TimeSeriesDataFrame, feature_name: str) -> TimeSeriesDataFrame:
+        if feature_name in self.covariate_metadata.static_features:
+            data.static_features[feature_name] = data.static_features[feature_name].sample(frac=1)
+        else:
+            data[feature_name] = data[feature_name].sample(frac=1)
+        return data
```

Comment:
BUG: To properly calculate importance that is comparable across features, the shuffling needs to be done with the same indices for each feature. This requires using a seed.

Otherwise, some features might just get lucky and get a favorable `.sample` order, which pollutes the correctness of the feature importance score.

Reply:
I'm surprised. In expectation over shuffles and considering that features are independent by assumption I would assume this effect to be negligible if any. This is also reflected in the pseudocode in our main [reference](https://explained.ai/rf-importance/#4). In the [eli5](https://github.com/eli5-org/eli5/blob/master/eli5/permutation_importance.py) implementation, it's only mentioned as a computational trick. It's also not in the Breiman paper.

Reply:
I agree that it probably doesn't make a large difference, but since we are scoring each of the features, we should ensure that the way we score them is aligned, so it doesn't hurt to use a fixed sample order. I'd strongly prefer doing a fixed order as it avoids strange scenarios (for example, if the user asks for the importance of features (A, B, D) and in a separate call features (A, B, C, D), they will get a different feature importance for D in the two calls, because it will use a different sample order.